### PR TITLE
feat: ConnectedParentFeed and ConnectedChildFeed

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

This PR does a lot in a small area of the app. 

It refactors the `HorizontalContentSeriesFeedConnected` component to rip out much of the core code, refactors it into a `HorizontalFeedConnected` component, and then uses that core code to add an additional `ContentParentFeed` component and `ContentChildFeed` component. 

I chose to go this route, rather than write the same code three times, because the core code that is shared can be tricky and prone to bugs. Having a smaller surface area helps us keep our potential surface area for issues down.  

I chose to create a separate component for `Parent` and `Children` because it's usually pretty obvious if a type has children or siblings. I wanted to reduce the overall complexity of the `HorizontalFeed` logic by having a separate component for each feed that do a single job (render siblings or children). 

### How do I test this PR?

As usual, stories :D Just note, that the pagination is a little wack because `loading` is never `true` when refetching with the mocks.

### What automated tests did you write?

100% coverage via jest snapshots. 

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apollos-ui-connected/src/ContentChildFeedConnected/ContentChildFeedConnected.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    updateQuery={(previousResult, { fetchMoreResult }) => {
+      const newEdges = get(
+        fetchMoreResult,
+        'node.siblingContentItemsConnection.edges',
+        []
+      );
+      return {
+        node: {
+          ...previousResult.node,
+          siblingContentItemsConnected: {
```
Comment (vinnyjth) [inline packages/apollos-ui-connected/src/ContentChildFeedConnected/ContentChildFeedConnected.js:32]:
```suggestion
          siblingContentItemsConnection: {
```

File: packages/apollos-ui-connected/src/ConnectedHorizontalFeed.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
```
Comment (redreceipt) [inline packages/apollos-ui-connected/src/ConnectedHorizontalFeed.js:1]: Shouldn't this component be called `HorizontalFeedConnected`?

File: packages/apollos-ui-connected/src/ContentChildFeedConnected/getContentChildSiblings.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import gql from 'graphql-tag';
+import ApollosConfig from '@apollosproject/config';
+
+export default gql`
+  query getContentChildSiblings($nodeId: ID!, $cursor: String) {
+    node(id: $nodeId) {
+      id
+      ... on ContentItem {
```
Comment (redreceipt) [inline packages/apollos-ui-connected/src/ContentChildFeedConnected/getContentChildSiblings.js:8]: this is the deprecated portion right?

**Comment by redreceipt** [comment]
@vinnyjth got this in the story
Screen Shot 2020-09-10 at 9 36 32 AM

**Comment by vinnyjth** [comment]
> @vinnyjth got this in the story
> 
> Screen Shot 2020-09-10 at 9 36 32 AM
> 
> 

This is to be expected due to how the mocks work. I left a comment in the PR description

File: packages/apollos-ui-connected/src/ConnectedHorizontalFeed.js (hunk 2 above)
Comment (vinnyjth) [reply packages/apollos-ui-connected/src/ConnectedHorizontalFeed.js:1]: Sure! I'll make that change

File: packages/apollos-ui-connected/src/ContentChildFeedConnected/getContentChildSiblings.js (hunk 3 above)
Comment (vinnyjth) [reply packages/apollos-ui-connected/src/ContentChildFeedConnected/getContentChildSiblings.js:8]: Yes and no. Technically not deprecated yet. Maybe in. 2.x version, but for now it's a totally valid approach.

**Review by redreceipt — APPROVED** [review]
left a question about the name of a component. feel free to merge if you don't agree. looks good!

**Review by redreceipt — APPROVED** [review]

**Pull request merged**